docs/proposals/diamond-design.md

```diff
@@ -0,0 +1,458 @@
+  ```
+* This does not break existing bundles
+
+
+#### Lineage tracking
+
+Premises:
+* We assume that usually, all diamonds are initiated and run by the same contributor (e.g. some ARGO workflow with a runID)
+* However, in the case of several contributors, each contributing some splits to the diamond,
+  we should keep track of all of them in the list of contributors to that bundle
+
+Design:
+* Each split keeps track of its originating contributor
+* This metadata is at first recorded in:
+  ```
+  /splits/{repo}/{diamond-id}/{split-id}/split.yaml
+  ```
+* At commit time, we merge contributors into the resulting bundle:
+  ```
+  /bundles/{repo}/{bundle-id}/bundle.yaml
+  ```
+
+## Additional commands
+
+### List all initiated diamonds on a repo
+```
+datamon bundle diamond list --repo {repo}
+```
+
+Completed diamonds are available in this listing until they are eventually cleanup by the background cleaner.
+
+### List all ongoing splits for a diamond on a repo
+```
+datamon bundle diamond split list --repo {repo} --diamond {diamond-id}
+```
+
+Report about the ID and status and timings of every ongoing or failed split.
+
+### List files in one or all splits
+```
+datamon bundle diamond split files --repo {repo} --diamond {diamond-id} [--split-id {split-id}]
+```
+
+### Report about conflicting files on a bundle
+```
+datamon bundle get --repo {repo} --bundle {bundle-id} --with-conflicts|--with-checkpoints
+```
+This will report about conflicting paths, or checkpointed versions in the case of incremental uploads.
+
+### Report about diamond-originated information on a bundle
+```
+datamon bundle get --repo {repo} --bundle {bundle-id} --format `{{ range .Diamonds}}{{ printf "%v" .}}{{ end }}`
+```
+or
+```
+datamon bundle get --repo {repo} --bundle {bundle-id} --with-diamonds
+```
+
+### Inspect conflicts
+
+```
+datamon bundle list files --repo {repo} --bundle {bundle-id} --with-conflicts|--with-checkpoints
+```
+
+This list files which experienced conflicts (resp. overwrites) during a diamond worklow (resp. an incremental update).
+
+```
+datamon bundle download --repo {repo} --bundle {bundle-id} --with-conflicts|--with-checkpoints --destination {local path}
+```
+
+This download the different conflicting (resp. checkpointed) versions of conflicting (resp. overwritten) files ({filename}-{split-id}).
+(filtering options apply like in `datamon bundle download`, so does the `datamon bundle download file` subcommand).
+
+Each downloaded file will be: either the original (winning) version, or suffixed split ID
+
+Example:
+```
+/destination/my-conflicting-file.data                                     #  <-- winner
+/destination/my-conflicting-file.data-0ujsszwN8NRY24YaXiTIE2VWDTS         #  <-- candidate
+/destination/my-conflicting-file.data-0ujsswThIGTUYm2K8FjOOfXtY1K         #  <-- candidate
+/destination/my-conflicting-file.data-0ujssxh0cECutqzMgbtXSGnjorm         #  <-- candidate
+```
+
+### Resolve conflicts
```

Comment:
Same as above, not in MVP, if we see developers battle this often, we can then decide how to help.
My suspicion is that developers might have requirements that we do not quite understand.

Reply:
Removed everything related to resolving conflicts.